L80-csharp-load-balancing.md

```diff
@@ -0,0 +1,394 @@
+    }
+    public abstract class LoadBalancerFactory {
+        protected LoadBalancerFactory();
+        public abstract string Name { get; }
+        public abstract LoadBalancer Create(LoadBalancerOptions options);
+    }
+    public sealed class PickContext {
+        public PickContext();
+        public HttpRequestMessage? Request { get; set; }
+    }
+    public sealed class PickFirstBalancerFactory : LoadBalancerFactory {
+        public PickFirstBalancerFactory();
+        public override string Name { get; }
+        public override LoadBalancer Create(LoadBalancerOptions options);
+    }
+    public sealed class PickResult {
+        public Status Status { get; }
+        public Subchannel? Subchannel { get; }
+        public PickResultType Type { get; }
+        public static PickResult ForDrop(Status status);
+        public static PickResult ForFailure(Status status);
+        public static PickResult ForQueue();
+        public static PickResult ForSubchannel(Subchannel subchannel, Action<CompletionContext>? onComplete = null);
+        public void Complete(CompletionContext context);
+    }
+    public abstract class Resolver : IDisposable {
+        protected Resolver();
+        public void Dispose();
+        public abstract void Refresh();
+        public abstract void Start(Action<ResolverResult> listener);
+        protected virtual void Dispose(bool disposing);
+    }
+    public abstract class ResolverFactory {
+        protected ResolverFactory();
+        public abstract string Name { get; }
+        public abstract Resolver Create(ResolverOptions options);
+    }
+    public sealed class ResolverResult {
+        public IReadOnlyList<BalancerAddress>? Addresses { get; }
+        public BalancerAttributes Attributes { get; }
+        public ServiceConfig? ServiceConfig { get; }
+        public Status? ServiceConfigStatus { get; }
+        public Status Status { get; }
+        public static ResolverResult ForFailure(Status status);
+        public static ResolverResult ForResult(IReadOnlyList<BalancerAddress> addresses);
+        public static ResolverResult ForResult(IReadOnlyList<BalancerAddress> addresses, ServiceConfig? serviceConfig, Status? serviceConfigStatus);
+    }
+    public sealed class RoundRobinBalancerFactory : LoadBalancerFactory {
+        public RoundRobinBalancerFactory();
+        public override string Name { get; }
+        public override LoadBalancer Create(LoadBalancerOptions options);
+    }
+    public sealed class StaticResolverFactory : ResolverFactory {
+        public StaticResolverFactory(Func<Uri, IEnumerable<BalancerAddress>> addressesCallback);
+        public override string Name { get; }
+        public override Resolver Create(ResolverOptions options);
+    }
+    public sealed class Subchannel : IDisposable {
+        public BalancerAttributes Attributes { get; }
+        public BalancerAddress? CurrentAddress { get; }
+        public ConnectivityState State { get; }
```

Comment:
```suggestion
```

Removed `State` property from subchannel in latest preview as it wasn't threadsafe. Load balancers should track state using OnStateChanged callback.